src/Courses/TopicGrades/GradingPage.tsx

```diff
@@ -25,248 +32,100 @@ export const TopicGradingPage: React.FC<TopicGradingPageProps> = () => {
-            setSelectedInfo({
-                path: currentPath,
-                seed: currentSeed,
-                problem: selected.problem,
-                grade: selectedInfo.grade,
-                workbooks: selectedInfo.workbooks,
-            });
+        if (_.isNil(params.topicId)) {
+            logger.error('topicId is null');
+            throw new Error('An unexpected error has occurred');
         } else {
-            setSelectedInfo({
-                problem: selected.problem,
-                grade: selectedInfo.grade,
-                workbooks: selectedInfo.workbooks,
-                workbook: selected.workbook,
-            });
+            fetchProblems(parseInt(params.topicId, 10));
         }
-    }, [selected.workbook]);
-
-    useEffect(() => {
-        logger.debug('GradingPage: overriding grade', gradeOverride.effectiveScore);
-        let currentMap = problemMap;
-        if (!_.isNil(selectedInfo.problem) &&
-            !_.isNil(currentMap[selectedInfo.problem.id])
-        ) {
-            // save our nil-check progress to avoid TS complaints with the following
-            let problem = currentMap[selectedInfo.problem.id];
-            if (!_.isNil(gradeOverride) &&
-                !_.isNil(gradeOverride.effectiveScore) &&
-                !_.isNil(problem.grades) &&
-                !_.isNil(selected.user) &&
-                !_.isNil(problem.grades[selected.user.id])
-            ) {
-                // edit the relevant problem dictionary
-                problem.grades[selected.user.id].effectiveScore = gradeOverride.effectiveScore;
-                // and put it back in the map
-                currentMap[selectedInfo.problem.id] = problem;
-                setProblemMap(currentMap);
-            }
-        }
-    }, [gradeOverride]);
+    }, [params.topicId, users]);
 
     const fetchProblems = async (topicId: number) => {
-        let currentProblems = problems;
-        // don't re-fetch the topic if we've already got it...
-        if (_.isEmpty(currentProblems) || currentProblems?.[0].courseTopicContentId !== params.topicId) {
-            const res = await getAssessmentProblemsWithWorkbooks({ topicId });
-            currentProblems = _(res.data.data.problems)
-                .map((p) => { return new NewProblemObject(p); })
+        // fetchProblems is only called when topicId changes ()
+        try {
+            const res = await getTopic({ id: topicId, includeQuestions: true });
+            const currentProblems = _(res.data.data.questions)
+                .map((p) => { return new ProblemObject(p); })
                 .sortBy(['problemNumber'], ['asc'])
                 .value();
             // currentProblems = _.map(currentProblems, (p) => {return new ProblemObject(p);});
             setProblems(currentProblems);
 
-            const currentTopic = res.data.data.topic;
+            const currentTopic = res.data.data as TopicObject;
```

Comment:
I think you should use `new TopicObject` instead of casting, because any logic we'll have on the class won't run otherwise.

Reply:
```suggestion
            const currentTopic = new TopicObject(res.data.data);
```